resolver-dns/src/main/java/io/netty/resolver/dns/DnsNameResolver.java

```diff
@@ -1133,15 +1146,58 @@ public void run() {
-                                       DnsCache resolveCache,
-                                       boolean completeEarlyIfPossible) {
+    private void doResolveAllUncached0(final String hostname,
+                                       final DnsRecord[] additionals,
+                                       final Promise<?> originalPromise,
+                                       final Promise<List<InetAddress>> promise,
+                                       final DnsCache resolveCache,
+                                       final boolean completeEarlyIfPossible) {
 
         assert executor().inEventLoop();
 
+        if (inflightLookups != null) {
+            if (additionals == null || additionals.length == 0) {
+                Future<List<InetAddress>> inflightFuture = inflightLookups.get(hostname);
+                if (inflightFuture != null) {
+                    inflightFuture.addListener(new GenericFutureListener<Future<? super List<InetAddress>>>() {
+                        @SuppressWarnings("unchecked")
+                        @Override
+                        public void operationComplete(Future<? super List<InetAddress>> future) {
+                            if (future.isSuccess()) {
+                                promise.setSuccess((List<InetAddress>) future.getNow());
+                            } else {
+                                Throwable cause = future.cause();
+                                if (isTimeoutError(cause)) {
+                                    // The failure was caused by a timeout. This might be happening as a result of
+                                    // the remote server to be overloaded for some short amount of time or because
+                                    // UDP packets were dropped on the floor. In this case lets try to just do the
+                                    // query explicit to not cascade this possible temporary failure.
+                                    final DnsServerAddressStream nameServerAddrs =
+                                            dnsServerAddressStreamProvider.nameServerAddressStream(hostname);
+                                    new DnsAddressResolveContext(DnsNameResolver.this, originalPromise,
+                                            hostname, additionals, nameServerAddrs,
+                                            maxQueriesPerResolve, resolveCache,
+                                            authoritativeDnsServerCache, completeEarlyIfPossible)
+                                            .resolve(promise);
+                                } else {
+                                    promise.setFailure(cause);
+                                }
+                            }
+                        }
+                    });
+                    return;
+                // Check if we have space left in the map.
+                } else if (maxNumConsolidation == 0 || inflightLookups.size() < maxNumConsolidation) {
```

Comment:
IIUC the expected behavior is as follows: 

```
<0 - not enabled
0 - always consolidate
>0 - consolidate that amount
```

do we need the special case for `0` and if so should qualify the javadocs?

Reply:
alternative combination that might be easier to understand:
```
0 - not enabled
>0 - consolidate that amount
Integer.MAX_VALUE - always consolidate
```

Reply:
I used the one that @idelpivnitskiy suggested